inc/Engine/Media/AboveTheFold/Frontend/Controller.php

```diff
@@ -336,4 +343,26 @@ private function is_mobile(): bool {
 			&& $this->options->get( 'do_caching_mobile_files', 0 )
 			&& wp_is_mobile();
 	}
+
+	/**
+	 * The `inject_beacon` function is used to inject a JavaScript beacon into the HTML content.
+	 *
+	 * @param string $html The HTML content where the beacon will be injected.
+	 *
+	 * @return string The modified HTML content with the beacon script injected just before the closing body tag.
+	 */
+	public function inject_beacon( $html ): string {
+		if ( ! $this->filesystem->exists( rocket_get_constant( 'WP_ROCKET_ASSETS_JS_PATH' ) . 'lcp-beacon.min.js' ) ) {
+			return $html;
+		}
+
+		// Get the URL of the script.
+		$script_url = rocket_get_constant( 'WP_ROCKET_ASSETS_JS_URL' ) . 'lcp-beacon.min.js';
```

Comment:
Couple of things for the script:
- we should have an unminified version too (can be useful for testing/debugging)
- we should be able to easily swap between the unminified/minified by changing the `min` value based on the `SCRIPT_DEBUG` constant value

Reply:
The unminified version is here : `assets/js/lcp-beacon.js`.

I've added a condition on the constant `SCRIPT_DEBUG` to load the unminified version while debugging.